src/App.tsx

```diff
@@ -15,36 +17,106 @@ export const goodsFromServer = [
+) {
+  const renderedGoods = [...goods];
+
+  renderedGoods.sort((good1, good2) => {
+    switch (action) {
+      case sortingActions.alphabetically:
+        return good1.trim().localeCompare(good2.trim());
+      case sortingActions.length:
+        return good1.trim().length - good2.trim().length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    renderedGoods.reverse();
+  }
+
+  return renderedGoods;
+}
+
+function resetState<T extends unknown[]>(
+  ...args: { [K in keyof T]: { setter: StateSetter<T[K]>; initialValue: T[K] } }
+): void {
+  args.forEach(({ setter, initialValue }) => setter(initialValue));
+}
+
 export const App: React.FC = () => {
+  const [sortingAction, setSortingAction] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const { alphabetically, length } = sortingActions;
+
+  const handledGoods = handleGoodsSortAction(
+    goodsFromServer,
+    sortingAction,
+    reversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortingAction !== alphabetically,
+          })}
+          onClick={() => setSortingAction(alphabetically)}
+        >
           Sort alphabetically
         </button>
-
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortingAction !== length,
+          })}
+          onClick={() => setSortingAction(length)}
+        >
           Sort by length
         </button>
-
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortingAction || reversed) && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={() =>
+              resetState(
+                { setter: setReversed, initialValue: false },
+                { setter: setSortingAction, initialValue: '' },
+              )
+            }
+          >
+            Reset
+          </button>
+        )}{' '}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {handledGoods.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, each element should have a unique `key` prop. Using 'good' as the key is risky if the list contains duplicate items. Consider using a combination of the item's content and its index, or another unique identifier.

Reply:
no intel to fix this in object